### PR TITLE
fix: beforeunload and unload firing in BrowserViews

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/20746.
Closes https://github.com/electron/electron/issues/27722.

Fixes an issue where `beforeunload` and `unload` weren't firing at all in `BrowserView`s. This was happening because we were only checking the BrowserWindow's `webContents` to see whether or not the events needed to be fired, and we also needed to check the `BrowserView`s WebContents.

However, this change makes it such that the event will be emitted for `BrowserViews` but will _not_ be respected - I've chosen not to tie the `BrowserView` lifecycle to its owning BrowserWindow should one exist per the [specification](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event). Either way this is undefined behavior, and the alternative would be to wait until _all_ associated `BrowserViews` for a `BrowserWindow` had their `webContents` destroyed, and given that a `BrowserView` doesn't _need_ to be associated with a `BrowserWindow` this would ultimately be more confusing for end users.

cc @zcbenz @deepak1556 @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where `beforeunload` and `unload` weren't firing properly in `BrowserView`s.

### CODE REVIEW

File: docs/api/web-contents.md [hunk 1]
```diff
@@ -375,6 +375,8 @@ win.webContents.on('will-prevent-unload', (event) => {
 })
 ```
 
+**Note:** This will be emitted for `BrowserViews` but will _not_ be respected - this is because we have chosen not to tie the `BrowserView` lifecycle to its owning BrowserWindow should one exist per the [specification](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event).
```
Comment (nornagon) [inline docs/api/web-contents.md:378]: what about this event won't be respected? preventing the default?

Comment (codebytere) [reply docs/api/web-contents.md:378]: @nornagon yes - the function body will be run but the return value won't be respected

**Review by zcbenz — APPROVED** [review]

File: docs/api/web-contents.md (hunk 1 above)
Comment (nornagon) [reply docs/api/web-contents.md:378]: Er, sorry for not getting back to this. I think this is unclear in the docs as written and should be clarified.

**Comment by pushkin-** [comment]
@codebytere Hi - do you have an idea of when this could be backported to 11?

**Pull request merged**